src/utilities/googleAnalytics.ts

```diff
@@ -38,3 +47,53 @@ export const getNotionalActionFromHintState = (hint: SwapFormSubmitButtonHintSta
             return "";
     }
 }
+
+export const pushEvent = (eventName: string, eventValue: string | number, sessionId?: string, amm?: AugmentedAMM, agent?: string) => {
+    // with context
+    if (amm) {
+        const pool = amm.protocol + (amm.rateOracle.protocolId === 5 || amm.rateOracle.protocolId === 6 ? "_borrow" : "" );
+        window.dataLayer.push({'event': eventName, 'value': eventValue, 'sessionId': sessionId, 'pool': pool, 'agent': agent});
+    } else { // without context
+        window.dataLayer.push({'event': eventName, 'value': eventValue, 'sessionId': sessionId});
+    }
+}
+
+export const pushTxSubmition = (
+    eventName: string, 
+    notional: number | undefined,
+    margin: number | undefined,
+    action: string,
+    sessionId: string,
+    amm: AugmentedAMM,
+    agent?: string | number,
+    failMessage?: string
+) => {
+    let agentType = "";
```

Comment:
if possible all mapping logic should be extracted away from the core utilities. The mapping is a logic closer to the business layer, not the tracking layer. We should avoid mixing these logics to prevent tech debt.

Reply:
created a new util function `getAgent.ts` and used it in the component logic at [commit](https://github.com/Voltz-Protocol/voltz-ui/pull/268/commits/9bb0958572bdde864d23200e7f12fd6c19910057)